pkg/dmi.go

```diff
@@ -0,0 +1,169 @@
+package pkg
+
+import (
+	"bufio"
+	"fmt"
+	"os/exec"
+	"strings"
+)
+
+type Entry = map[string][]string
+type DMIMap = map[string]Entry
+
+type DMIDecoder struct {
+	installed  bool
+	cmdOutput  string
+	decodedMap DMIMap
+}
+
+// NewCheckerService creates new instance from the parser
+func NewDMIDecoder() DMIDecoder {
```

Comment:
We could have separated the concerns by passing string as a parameter, and we make it the user problem to get a valid dmidecode result to pass to the function, maybe using another function like `GetDMIDecodeOutput()`